VictoriaAlbert/VictoriaAlbert/VictoriaAlbertTableViewController.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  VictoriaAlbertTableViewController.swift
+//  VictoriaAlbert
+//
+//  Created by Tom Seymour on 11/10/16.
+//  Copyright © 2016 C4Q-3.2. All rights reserved.
+//
+
+import UIKit
+
+class VictoriaAlbertTableViewController: UITableViewController, UITextFieldDelegate {
+    
+    var museumThings: [MuseumThing] = []
+    
+    let victoriaAlbertCellIdentifyer = "VictoriaAlbertCell"
+    let vicAlbDetailSegueIdentifyer = "VictoriaAlbertDetailSegue"
+    
+    var vicAlbAPIEndpoint = "http://www.vam.ac.uk/api/json/museumobject/search?q=ring"
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadTableViewData()
+    }
+    
+    func loadTableViewData() {
+        APIHelper.manager.getData(endPoint: vicAlbAPIEndpoint) { (data: Data?) in
+            guard let unwrappedData = data else { return }
+            self.museumThings = MuseumThing.buildMuseumThingArray(from: unwrappedData)!
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        let allTextFieldText = textField.text ?? "ring"
+        textField.text = nil
+        let searchWord = allTextFieldText.replacingOccurrences(of: " ", with: "%20")
+        self.vicAlbAPIEndpoint = "http://www.vam.ac.uk/api/json/museumobject/search?q=" + searchWord
+        loadTableViewData()
+        
+        return true
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return museumThings.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: victoriaAlbertCellIdentifyer, for: indexPath)
+        let thisMuseumThing = museumThings[indexPath.row]
+        cell.textLabel?.text = "\(indexPath.row + 1): \(thisMuseumThing.object), \(thisMuseumThing.dateMade), \(thisMuseumThing.placeMade)"
+        cell.detailTextLabel?.text = thisMuseumThing.title
+        
+        APIHelper.manager.getData(endPoint: thisMuseumThing.thumbImageString) { (data: Data?) in
+            
+            guard let unwrappedData = data else { return }
+            DispatchQueue.main.async {
+                cell.imageView?.image = UIImage(data: unwrappedData)
+                cell.setNeedsLayout()
+            }
+        }
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```

Comment:
We do not need to implement both didSelectRowAt and have a segue. I see what you did is kind of clever, passing the object you want and so I can't complain too much. But the segue alone would be enough in the story board if you cast the sender as a UITableViewCell and pass it to the tableView in ```prepare(for:)``` to get the indexPath. With the index path you can look up the data from the array.